integration/deno-compiler-test.ts

```diff
@@ -0,0 +1,231 @@
+          const data = useLoaderData();
+          return (
+            <ul>
+              <li>{fake}</li>
+
+              <li>{urlComponent}</li>
+              <li>{urlUtil()}</li>
+              <li>{data.a}</li>
+              <li>{data.b}</li>
+
+              <li>{npmComponent}</li>
+              <li>{npmUtil()}</li>
+              <li>{data.c}</li>
+              <li>{data.d}</li>
+            </ul>
+          )
+        }
+      `,
+      "node_modules/npm-component/package.json": json({
+        name: "npm-component",
+        version: "1.0.0",
+        sideEffects: false,
+      }),
+      "node_modules/npm-component/index.js": js`
+        module.exports = { npmComponent: () => "NPM_COMPONENT" };
+      `,
+      "node_modules/npm-util/package.json": json({
+        name: "npm-util",
+        version: "1.0.0",
+        sideEffects: false,
+      }),
+      "node_modules/npm-util/index.js": js`
+        module.exports = { npmUtil: () => "NPM_UTIL" };
+      `,
+      "node_modules/npm-server-only/package.json": json({
+        name: "npm-server-only",
+        version: "1.0.0",
+        sideEffects: false,
+      }),
+      "node_modules/npm-server-only/index.js": js`
+        module.exports = { npmServerOnly: () => "NPM_SERVER_ONLY" };
+      `,
+      "node_modules/deno-pkg/package.json": json({
+        name: "deno-pkg",
+        version: "1.0.0",
+        type: "module",
+        main: "./default.js",
+        exports: {
+          deno: "./deno.js",
+          worker: "./worker.js",
+          default: "./default.js",
+        },
+        sideEffects: false,
+      }),
+      "node_modules/deno-pkg/deno.js": js`
+        export default "DENO_EXPORTS";
+      `,
+      "node_modules/deno-pkg/worker.js": js`
+        export default "WORKER_EXPORTS";
+      `,
+      "node_modules/deno-pkg/default.js": js`
+        export default "DEFAULT_EXPORTS";
+      `,
+    },
+  });
+});
+
+test("compiler does not bundle url imports for server", async () => {
+  let serverBundle = await fse.readFile(findServerBundle(projectDir), "utf8");
+  expect(serverBundle).toMatch(importPattern("https://deno.land/x/util.ts"));
+  expect(serverBundle).toMatch(
+    importPattern("https://deno.land/x/server-only.ts")
+  );
+
+  // server-side rendering
+  expect(serverBundle).toMatch(
+    importPattern("https://deno.land/x/component.ts")
+  );
+});
+
+test("compiler does not bundle url imports for browser", async () => {
+  let browserBundle = findBrowserBundle(projectDir);
+  let browserCodeFiles = await findCodeFiles(browserBundle);
+
+  let utilFiles = await searchFiles(
+    importPattern("https://deno.land/x/util.ts"),
+    browserCodeFiles
+  );
+  expect(utilFiles.length).toBeGreaterThanOrEqual(1);
+
+  let componentFiles = await searchFiles(
+    importPattern("https://deno.land/x/component.ts"),
+    browserCodeFiles
+  );
+  expect(componentFiles.length).toBeGreaterThanOrEqual(1);
+
+  /*
+  url imports _could_ have side effects,
```

Comment:
Let's add `sideEffects: false` in our plugin that resolves HTTP imports since it is the most common thing for http imports to not have side effects. https://esbuild.github.io/plugins/#on-resolve-results